pkg/api/datasources.go

```diff
@@ -598,19 +600,27 @@ func getEncodedString(jsonData *simplejson.Json, key string) string {
 	return string(val)
 }
 
-func checkTeamHTTPHeaderPermissions(hs *HTTPServer, c *contextmodel.ReqContext, ds *datasources.DataSource, cmd datasources.UpdateDataSourceCommand) (bool, error) {
+func checkTeamHTTPHeadersDiff(ds *datasources.DataSource, jsonData *simplejson.Json) bool {
+	if ds == nil || jsonData == nil {
```

Comment:
I think there are two issues with this check:
1. this will always return `false` when a new DS is created, as `ds == nil` for [that call](https://github.com/grafana/grafana/pull/93496/files#diff-018aba0eacc641714beb1e80c74d5fb002e6b47a397f6595553872e290a8be54R457);
2. if team HTTP headers already exist on a DS, we allow clearing them (by setting `jsonData` to `nil`).

Should this be `if ds == nil && jsonData == nil` instead, plus a separate logic to handle teamHTTP headers set during DS creation?

Reply:
> this will always return false when a new DS is created, as ds == nil
This is very true! I have to fix this.

> if team HTTP headers already exist on a DS, we allow clearing them (by setting jsonData to nil).
they should not be allowed to clear them by using a `jsonData` `nil`

We only allow them to be deleted through the other api endpoint.

Reply:
not relevant if we go with the strategy of overriding the previous Rules